internal/cmd/ske/credentials/kubeconfig/login/login.go

```diff
@@ -0,0 +1,223 @@
+				// cert expired, request new
+				_ = cache.DeleteObject(model.CacheKey)
+				return getCacheAndOutputKubeconfig(ctx, cmd, apiClient, model, false, nil)
+			} else if time.Now().Add(time.Minute * 15).After(certificate.NotAfter.UTC()) {
+				// cert expires in 15min, refresh
+				return getCacheAndOutputKubeconfig(ctx, cmd, apiClient, model, true, kubeconfig)
+			}
+
+			if err := output(cmd, model.CacheKey, kubeconfig); err != nil {
+				return err
+			}
+			return nil
+		},
+	}
+	return cmd
+}
+
+func parseInput() (*inputModel, error) {
+	obj, _, err := exec.LoadExecCredentialFromEnv()
+	if err != nil {
+		return nil, fmt.Errorf("LoadExecCredentialFromEnv: %w", err)
+	}
+
+	if err := clientauthenticationv1.AddToScheme(scheme.Scheme); err != nil {
+		return nil, err
+	}
+
+	obj, err = scheme.Scheme.ConvertToVersion(obj, clientauthenticationv1.SchemeGroupVersion)
+	if err != nil {
+		return nil, fmt.Errorf("ConvertToVersion: %w", err)
+	}
+
+	execCredential, ok := obj.(*clientauthenticationv1.ExecCredential)
+	if !ok {
+		return nil, fmt.Errorf("Conversion to ExecCredential failed")
+	}
+	if execCredential == nil || execCredential.Spec.Cluster == nil {
+		return nil, fmt.Errorf("ExecCredential contains not all needed fields")
+	}
+	config := &SKEClusterConfig{}
+	err = json.Unmarshal(execCredential.Spec.Cluster.Config.Raw, config)
+	if err != nil {
+		return nil, fmt.Errorf("unmarshal: %w", err)
+	}
+
+	return &inputModel{
+		ClusterName: config.ClusterName,
+		ProjectId:   config.STACKITProjectID,
+		CacheKey:    fmt.Sprintf("ske-login-%x", sha256.Sum256([]byte(execCredential.Spec.Cluster.Server))),
+	}, nil
+}
+
+func buildRequest(ctx context.Context, apiClient *ske.APIClient, model *inputModel) ske.ApiCreateKubeconfigRequest {
+	req := apiClient.CreateKubeconfig(ctx, model.ProjectId, model.ClusterName)
+	expirationSeconds := "1800" // 30 min
+
+	return req.CreateKubeconfigPayload(ske.CreateKubeconfigPayload{ExpirationSeconds: &expirationSeconds})
+}
+
+func parseKubeConfigToExecCredential(kubeconfig *rest.Config) (*clientauthenticationv1.ExecCredential, error) {
+	certPem, _ := pem.Decode(kubeconfig.CertData)
+	if certPem == nil {
+		return nil, fmt.Errorf("login SKE kubeconfig")
+	}
+
+	certificate, err := x509.ParseCertificate(certPem.Bytes)
+	if err != nil {
+		return nil, fmt.Errorf("login SKE kubeconfig: %w", err)
+	}
+
+	outputExecCredential := clientauthenticationv1.ExecCredential{
+		TypeMeta: v1.TypeMeta{
+			APIVersion: clientauthenticationv1.SchemeGroupVersion.String(),
+			Kind:       "ExecCredential",
+		},
+		Status: &clientauthenticationv1.ExecCredentialStatus{
+			ExpirationTimestamp:   &v1.Time{Time: certificate.NotAfter.Add(-time.Minute * 15)},
+			ClientCertificateData: string(kubeconfig.CertData),
+			ClientKeyData:         string(kubeconfig.KeyData),
+		},
+	}
+	return &outputExecCredential, nil
+}
+
+func getCachedKubeConfig(key string) *rest.Config {
+	cachedKubeconfig, err := cache.GetObject(key)
+	if err != nil {
```

Comment:
we will introduce debug logs soon, at that point we should log these errors

Reply:
We could add them, but because this command will only be invoked by `kubectl` users won't ever see them. In this case, a log file would be the better approach,

Reply:
The log file is something that we could have across the CLI, so it is not specific for this command. We can add it later when that is implemented